js/ui/scheduler/ui.scheduler.recurrence_editor.js

```diff
@@ -362,9 +364,9 @@ const RecurrenceEditor = Editor.inherit({
 
         this._daysOfWeek = [];
 
-        for(let i = 0; i < 7; i++) {
+        for(let i = 0; i < DAYS_IN_WEEK; i++) {
             const daysOffset = this._getFirstDayOfWeek() + i;
-            const dayIndex = daysOffset % 7;
+            const dayIndex = daysOffset % DAYS_IN_WEEK;
```

Comment:
There is 3 copy-paste in the file. Lines: 369, 401, 876. It can be in the extra function. Something like this:
```
function getDayIndex() {
  const firstDay = this._getFirstDayOfWeek();
  return function(index) {
      return (index + firstDay ) % DAYS_IN_WEEK;
   }
}
```